components/nimbus/ios/Nimbus/NimbusCreate.swift

```diff
@@ -0,0 +1,86 @@
+}
+
+extension Nimbus {
+    /**
+     *  Create an instance of `Nimbus`.
+     *
+     * - Parameters:
+     *     - server: the server that experiments will be downloaded from
+     *     - appSettings: the name and channel for the app
+     *     - dbPath: the path on disk for the database
+     *     - enabled: intended for FeatureFlags. If false, then return a dummy `Nimbus` instance. Defaults to `true`.
+     *     - errorReporter: a closure capable of reporting errors. Defaults to using a logger.
+     */
+    public static func create(_ server: NimbusServerSettings?,
+                              appSettings: NimbusAppSettings,
+                              dbPath: String,
+                              enabled: Bool = true,
+                              errorReporter: @escaping NimbusErrorReporter = defaultErrorReporter) -> NimbusApi {
+        guard enabled else {
+            return NimbusDisabled()
+        }
+
+        let context = Nimbus.buildExperimentContext(appSettings)
+        let remoteSettings = server.map { server -> RemoteSettingsConfig in
+            let url = server.url.absoluteString
+            return RemoteSettingsConfig(
+                serverUrl: url,
+                bucketName: EXPERIMENT_BUCKET_NAME,
+                collectionName: EXPERIMENT_COLLECTION_NAME
+            )
+        }
+
+        do {
+            let nimbusClient = try NimbusClient(
+                appCtx: context,
+                dbpath: dbPath,
+                remoteSettingsConfig: remoteSettings,
+                // The "dummy" field here is required for obscure reasons when generating code on desktop,
+                // so we just automatically set it to a dummy value.
+                availableRandomizationUnits: AvailableRandomizationUnits(clientId: nil, dummy: 0)
+            )
+
+            return Nimbus(nimbusClient: nimbusClient, errorReporter: errorReporter)
+        } catch {
+            errorReporter(error)
+            return NimbusDisabled()
+        }
+    }
+
+    public static func buildExperimentContext(_: NimbusAppSettings,
+                                              bundle: Bundle = Bundle.main,
+                                              device: UIDevice = .current) -> AppContext {
+        let info = bundle.infoDictionary ?? [:]
+        return AppContext(appId: info["CFBundleIdentifier"] as? String ?? "unknown",
+                          appVersion: info["CFBundleShortVersionString"] as? String,
+                          appBuild: info["CFBundleVersion"] as? String,
+                          architecture: nil,
+                          deviceManufacturer: Sysctl.manufacturer,
+                          deviceModel: Sysctl.model,
+                          locale: Locale.current.identifier,
+                          os: device.systemName,
+                          osVersion: device.systemVersion,
+                          androidSdkVersion: nil,
+                          debugTag: "Nimbus.rs")
```

Comment:
Attention iOS people and Glean people.

Can we get the architecture easily? Why is `androidSdkVersion` so named?

Reply:
It is so named because it is an artifact of telemetry embedded in Glean that probably shouldn't be inside of Glean that is Android specific. It cargo culted its way into Nimbus from the initial copying of some of the Mako design. It should probably be removed as I don't recall it ever being used in Mako, or it could be modified to be the Nimbus SDK version or something worth targeting on, if we even need it.

Reply:
Here is a link to where Glean is fetching much of this same info: https://github.com/mozilla/glean/blob/bfa108f2b8d81e8cc7e19b05421baab07301c8d5/glean-core/ios/Glean/Glean.swift#L211

Reply:
`architechture` being `Sysctl.machine` and there is some massaging of the Locale going on to get it into `EN-US, EN-GB, DE-DU, etc...` format that Glean does.

Reply:
This is really helpful, thank you.